### PR TITLE
ci: build and push the docker image to ghcr

### LINKED ISSUE #54: Slow to start perhaps because this is a Docker container action?
Hi community,

I love this actions, but it takes almost 1 minutes to build the image to start the actions

image

Since one of our use case is for documentation checking, and it's a little bit slow for this use case.
How can I optimize it to make it start faster?

### PR DESCRIPTION
**Pull request by jizusun**
close #54

### CODE REVIEW

**Review by levonet — CHANGES_REQUESTED** [review]
I want to warn you. Publishing a Docker image and using it from a user repository is not acceptable from a security perspective.
Imagine that I use this action in a closed company repository. Running an opaque container can be used to access closed source code and compromise it.

Community tools should be as transparent as possible from a security point of view.
In this case, an opaque chain with the Docker image is added, which can fall under the control of an attacker.

**Comment by jizusun** [comment]
> I want to warn you. Publishing a Docker image and using it from a user repository is not acceptable from a security perspective. Imagine that I use this action in a closed company repository. Running an opaque container can be used to access closed source code and compromise it.
> 
> Community tools should be as transparent as possible from a security point of view. In this case, an opaque chain with the Docker image is added, which can fall under the control of an attacker.

Yes, I understand there is a risk in the software supply chain.
But I'm wondering what's the other solution to solve the problem I mentioned in this issue https://github.com/reviewdog/action-detect-secrets/issues/54, that is the actions is slow to start. 

This is why I didn't update the README in this PR.

If somebody want to skip the action build step ( 50s -> 9s), he can check the GHCR to use the prebuilt image after inspecting  whole the built process (all available in open source, no hidden part) rather than build and push by himself, also just like here https://github.com/reviewdog/action-actionlint uses GHCR. 

If somebody is still concerned about the integrity of the docker image, he just used the current way, that is, building the docker image when using the action.

Do you think it is okay for you to accept the PR , and I'll create a follow-up PR to resolve your concern.
My plan is
- Clarify the supply chain problem
- Warn the user the risk
- The current one as the recommend one, and the pre-built approach as an alternative one

**Comment by jizusun** [comment]
@shogo82148  Can you help to approve the workflow to run?
Sorry that previous workflow failed because of `gcc`

**Review by shogo82148 — APPROVED** [review]

File: .github/workflows/docker-build.yml
Comment (javierjulio) [inline .github/workflows/docker-build.yml:1]: Let's remove this file for now. We don't need to do this to improve the action run time.

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-
-RUN set -eux \
-    && apt-get update \
-    && apt-get install -y --no-install-recommends \
-        git \
-        wget \
+RUN apk --no-cache add git gcc musl-dev \
     && wget -O - -q https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh| sh -s -- -b /usr/local/bin/ ${REVIEWDOG_VERSION} \
-    && pip install detect-secrets[word_list]
+    && pip install detect-secrets[word_list]
```
Comment (javierjulio) [inline Dockerfile:5]: Avoid unnecessary changes. This space was added here. 

```suggestion
    && pip install detect-secrets[word_list]
```

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,8 @@
...
-
+FROM python:3.11.4-alpine
 ENV REVIEWDOG_VERSION=v0.17.0
-
-RUN set -eux \
-    && apt-get update \
-    && apt-get install -y --no-install-recommends \
-        git \
-        wget \
+RUN apk --no-cache add git gcc musl-dev \
```
Comment (javierjulio) [inline Dockerfile:3]: Was gcc and musl-dev only added for the docker push workflow? If so, then let's make sure to remove these, please.

```suggestion
RUN apk --no-cache add git \
```

**Review by javierjulio — CHANGES_REQUESTED** [review]
@jizusun thank you. I agree with @levonet we should avoid pushing the image for now. There is no need to do this as by just using alpine we should see a faster start time in workflow runs. I encountered the same slowness at times. I would accept just the alpine change for now.

**Comment by haya14busa** [comment]
Can somebody create a pr to change the image to alpine or slim?

**Comment by javierjulio** [comment]
Yes, a new PR was created in #71 which looks good.

**Pull request closed without merging**